libraries/chain/block.cpp

```diff
@@ -111,4 +112,22 @@ namespace eosio { namespace chain {
       return validate_and_extract_block_extensions( block_extensions );
    }
 
+   signed_block_ptr pruned_block::to_signed_block() const {
+      if (prune_state != prune_state_type::complete_legacy)
+         return signed_block_ptr{};
+
+      auto result = std::make_shared<signed_block>(*this);
+      result->block_extensions = this->block_extensions;
+
+      auto visitor = overloaded{
+         [](const transaction_id_type &id) -> transaction_receipt::trx_type { return id; },
+         [](const pruned_transaction &trx) -> transaction_receipt::trx_type { 
+            return packed_transaction{trx.get_signed_transaction(), trx.get_compression()}; 
```

Comment:
This would recompress and possibly give different a different `packed_trx` or `packed_context_free_data` than the original. We need to get the original packed forms of those, which should be possible for all `pruned_transaction` transactions within the block since we know that this block is `complete_legacy`.

Note that this will mean adding a way to access the `packed_context_free_data` within a `pruned_transaction` when it contains `full_legacy`.